NeoMathEngine/src/GPU/CUDA/CudaDevice.h

```diff
@@ -37,22 +37,25 @@ struct CCudaDevice : public CCrtAllocatedObject {
 	int ThreadMaxCount;
 	dim3 ThreadMax3DCount;
 	int WarpSize;
-	void* Handles[CUDA_DEV_SLOT_COUNT];
+	void* Handle;
 
-	CCudaDevice( int deviceNumber, size_t memoryLimit );
+	CCudaDevice( int deviceNumber, size_t memoryLimit, void* handle );
```

Comment:
Let's try to delete any constructors and use a new function CreateDevice. CreateDevice will capture slots and create a new CCudaDevice.

Reply:
Fixed